policy/policy.go

```diff
@@ -139,11 +125,110 @@ func ParsePolicy(str string) (Policy, error) {
 	return Policy{resolution: resolution, retention: retention}, nil
 }
 
-// NewPoliciesFromSchema creates multiple new policues from given schema policies.
-func NewPoliciesFromSchema(policies []*schema.Policy) ([]Policy, error) {
-	res := make([]Policy, 0, len(policies))
+// UnaggregatedPolicy represents a policy with a list of custom aggregation types
+type UnaggregatedPolicy struct {
+	policy             Policy
+	compressedAggTypes CompressedAggregationTypes
+}
+
+// NewUnaggregatedPolicyFromSchema creates a new unaggregated policy from a schema policy.
+func NewUnaggregatedPolicyFromSchema(p *schema.Policy) (UnaggregatedPolicy, error) {
+	if p == nil {
+		return EmptyUnaggregatedPolicy, errNilPolicySchema
+	}
+	precision := time.Duration(p.Resolution.Precision)
+	unit, err := xtime.UnitFromDuration(precision)
+	if err != nil {
+		return EmptyUnaggregatedPolicy, err
+	}
+
+	compressedAggTypes, err := NewCompressedAggregationTypesFromSchema(p.AggregationTypes)
+	if err != nil {
+		return EmptyUnaggregatedPolicy, err
+	}
+
+	return NewUnaggregatedPolicy(Policy{
```

Comment:
nit: you can call `NewPolicy` here instead of directly accessing the internal fields.